barretenberg/cpp/src/barretenberg/honk/proof_system/protogalaxy_prover.hpp

```diff
@@ -278,8 +285,32 @@ template <class ProverInstances_> class ProtoGalaxyProver_ {
             Utils::add_nested_tuples(univariate_accumulators, accumulators);
         }
         // Batch the univariate contributions from each sub-relation to obtain the round univariate
-        return Utils::template batch_over_relations<ExtendedUnivariateWithRandomization>(
-            univariate_accumulators, alpha, pow_univariate);
+        return Utils::template batch_over_relations<ExtendedUnivariateWithRandomization>(univariate_accumulators,
+                                                                                         alpha);
+    }
+
+    static Univariate<FF,
+                      (Flavor::BATCHED_RELATION_TOTAL_LENGTH - 1) * (ProverInstances::NUM - 1) + 1,
+                      ProverInstances::NUM>
+    compute_combiner_quotient(FF compressed_perturbator, ExtendedUnivariateWithRandomization combiner)
+    {
+        std::array<FF, (Flavor::BATCHED_RELATION_TOTAL_LENGTH - 2) * (ProverInstances::NUM - 1)>
+            combiner_quotient_evals = {};
+
+        // we need to evaluate at values not part in the vanishing set
+        for (size_t point = ProverInstances::NUM; point < combiner.size(); point++) {
+            auto idx = point - ProverInstances::NUM;
+            auto lagrange_0 = FF(1) - FF(point);
+            auto vanishing_polynomial = FF(point) * (FF(point) - 1);
+            combiner_quotient_evals[idx] =
+                (combiner.value_at(point) - compressed_perturbator * lagrange_0) / vanishing_polynomial;
```

Comment:
Could you implement this as a batch inversion or create an issue marking this as a future optimization?